### DISCUSSION TITLE
Creating a test for replacing spaces in answer

### DISCUSSION THREAD

**Discussion by smiechuskce** [DevAdventCalendar/DevAdventCalendar#415] [Q&A] [answered]
I paste here what I wrote in #252. 

I've coded the simple replace logic, that's not a problem. The problem is, I'm not really sure what and where to write a test for it. Currently, the whole logic takes just a single line of code, no many more after changes. Can't test this logic itself, because it's not separated from the controller. I could do that but there is another thought - does it make sense? It will be used only in this case, so there is no point creating another class for just this one method doing that simple logic.

I tried also to modify controllers test, just to see what happens if I'll change the answers from sample data. Those tests however pass even without changing logic to not remove spaces. I've checked out this line of TestControllerTests.cs

`207: this._testServiceMock.Setup(x => x.VerifyTestAnswer(It.IsAny<string>(), It.IsAny<List<string>>())).Returns(true);`

I'm a bit confused, because, assuming the test Index_CorrectAnswer_ReturnsAnswerViewModel tests how the controller method will behave if user pass a correct answer. But because of this line above this test will always pass, even if there will be a completely wrong answer passed. I think it doesn't test all the logic controller does for validating the correct answer.

@DevAdventCalendar/dac-mentoring-group Please correct me if I'm wrong.

**Comment by WTobor** [comment, marked as answer]
@smiechuskce please add a draft of PR - let's see some code :)

**Reply by WTobor** [reply]
I think that logic should be moved to Services - we don't want to have logic in Controllers. For now, it can be only one class.

About the test: yes, for that test only, mock will always return true; of course, we can set correct parameters there - it is not a problem. But it works only per test method, so it will not affect your test method.
